Orfium/toolbox
# [DEV-103][DEV-131] feat: add some state indicators for errors and loading

## Description



This PR introduces two new things
* a new loading indicator before the app loads to show to the user the loading state of the app
* a new global error handling for various steps between our state and auth0

## Screenshot
This shows both the loading state and an error showcase
![Screen Recording 2022-10-25 at 4 56 20 PM](https://user-images.githubusercontent.com/6433679/197794117-46e262ce-13d7-4283-bfa0-97a8545b5cd2.gif)



## Test Plan
* Check several user cases on TopBar like undefined etc
* Check various errors on a variety of vital functions across auth0 initialization
* Check if the error boundary renders correctly
